modules/http/http.c

```diff
@@ -299,6 +307,135 @@ _format_stats_instance(LogThreadedDestDriver *s)
   return stats;
 }
 
+static const gchar *
+_format_auth_header_name(const LogPipe *s)
+{
+  static gchar auth_header_name[1024];
+  const gchar *persist_name = _format_persist_name(s);
+  g_snprintf(auth_header_name, sizeof(auth_header_name), "%s.auth_header", persist_name);
+
+  return auth_header_name;
+}
+
+static gboolean
+_load_auth_header_from_persist_file(LogPipe *s)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *)s;
+  GlobalConfig *cfg = log_pipe_get_config(s);
+
+  gsize size;
+  guint8 version;
+  const gchar *persist_name = _format_auth_header_name(s);
+  gchar *auth_header_str = persist_state_lookup_string(cfg->state, persist_name, &size, &version);
+
+  if (!auth_header_str)
+    return FALSE;
+
+  gboolean ret = http_auth_header_load_from_string(self->auth_header, auth_header_str);
+  g_free(auth_header_str);
+
+  return ret;
+}
+
+static void
+_save_auth_header_to_persist_file(LogPipe *s)
+{
+  HTTPDestinationDriver *self = (HTTPDestinationDriver *)s;
+  GlobalConfig *cfg = log_pipe_get_config(s);
+  const gchar *persist_name = _format_auth_header_name(s);
+  const gchar *auth_header_str = http_auth_header_get_as_string(self->auth_header);
+
+  persist_state_alloc_string(cfg->state, persist_name, auth_header_str, -1);
+}
+
+static gboolean
+_auth_header_renew(HTTPDestinationDriver *self)
+{
+  gboolean ret = TRUE;
+
+  if (self->auth_header && http_auth_header_has_expired(self->auth_header))
```

Comment:
`http_auth_header_has_expired` is optional to implement, without a has_expired() method it will be always FALSE.
This means that we cannot force a renew on error.

We should either:
- make `has_expired` mandatory
- or create a separate function, e.g. `_auth_header_force_renew`.

Reply:
@gaborznagy : indeed, but renewal should be always explicitly tested
Making it as mandatory method is a valid option.

Reply:
Thanks, we can open a new PR later with asserting `has_expired` exists. :)